Source/WebCore/rendering/StyledMarkedText.cpp

```diff
@@ -203,4 +302,22 @@ Vector<StyledMarkedText> StyledMarkedText::coalesceAdjacentWithEqualDecorations(
     });
 }
 
+void StyledMarkedText::removeOverlappingForegroundRanges(Vector<StyledMarkedText>& markedTexts)
+{
+    Vector<std::pair<unsigned, unsigned>> ranges;
+    for (auto& markedText : markedTexts) {
+        if (!markedText.highlightName.isNull())
+            ranges.appendIfNotContains(std::make_pair(markedText.startOffset, markedText.endOffset));
+    }
+    // Check for non-highlight StyledMarkedTexts that overlap, if so then remove them.
+    unsigned index = 0;
+    while (index < markedTexts.size()) {
```

Comment:
this can be a normal for loop

Reply:
I don't increment the index if I remove something though, can it still be a normal for loop?

Reply:
ah, my bad again